src/win/thread.c

```diff
@@ -191,7 +191,13 @@ int uv_thread_create(uv_thread_t *tid, void (*entry)(void *arg), void *arg) {
 
 
 uv_thread_t uv_thread_self(void) {
-  return (uv_thread_t) uv_key_get(&uv__current_thread_key);
+  void* self;
+  uv_once(&uv__current_thread_init_guard, uv__init_current_thread_key);
+  self = uv_key_get(&uv__current_thread_key);
+  if (self == NULL)
+      /* This thread wasn't created with uv_thread_create. */
+      self = GetCurrentThread();
```

Comment:
Yes, it would be NULL for any thread not created by libuv (including the main thread). My point is that GetCurrentThread() returns a constant (I think it's -2), so NULL is just as good. 

In order to get a unique per-thread HANDLE for those threads that were not created by libuv, we would have to call DuplicateHandle on that pseudo-handle, but then we would also have to make sure we close those handles when the thread exits. AFAIK, no one has complained so far about libuv returning the same value for all threads not created by uv_thread_create().